### PR TITLE
Determine conntrack status based on connection info

### PR DESCRIPTION
**Pull request by akarpz**
### What does this PR do?

Allows us to stop offset guessing for the status field on the conntrack object, instead determining if NAT is happening in the relevant kprobes by inspecting the origin and reply connection info

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/ebpf/c/prebuilt/conntrack.h [hunk 1]
```diff
@@ -29,10 +29,15 @@ offset_ct(status)
...
-static __always_inline u32 ct_status(const struct nf_conn *ct) {
-    u32 status = 0;
-    bpf_probe_read_kernel_with_telemetry(&status, sizeof(status), (char*)ct + offset_ct_status());
-    return status;
+#define RETURN_IF_NOT_NAT(orig, reply)                      \
+    if (!is_conn_nat(orig, reply)) {                        \
+        return 0;                                           \
+    }
+
+bool is_conn_nat(conntrack_tuple_t orig, conntrack_tuple_t reply) {
```
Comment (hmahmood) [inline pkg/network/ebpf/c/prebuilt/conntrack.h:37]: Should `static __always_inline` this

Comment (hmahmood) [reply pkg/network/ebpf/c/prebuilt/conntrack.h:37]: Also should pass the args by pointer (preferably `const conntrack_tuple_t*`)

File: pkg/network/ebpf/c/prebuilt/conntrack.h [hunk 2]
```diff
@@ -29,10 +29,15 @@ offset_ct(status)
 offset_ct(netns)
 offset_ct(ino)
 
-static __always_inline u32 ct_status(const struct nf_conn *ct) {
-    u32 status = 0;
-    bpf_probe_read_kernel_with_telemetry(&status, sizeof(status), (char*)ct + offset_ct_status());
-    return status;
+#define RETURN_IF_NOT_NAT(orig, reply)                      \
```
Comment (hmahmood) [inline pkg/network/ebpf/c/prebuilt/conntrack.h:32]: You should move this macro and the function below to a common header file.

File: pkg/network/ebpf/c/runtime/conntrack.h [hunk 3]
```diff
@@ -28,10 +28,15 @@ static __always_inline u32 get_netns(void *p_net) {
     return net_ns_inum;
 }
 
-static __always_inline u32 ct_status(const struct nf_conn *ct) {
-    u32 status = 0;
-    bpf_probe_read_kernel_with_telemetry(&status, sizeof(status), (void *)&ct->status);
-    return status;
+#define RETURN_IF_NOT_NAT(orig, reply)                      \
```
Comment (hmahmood) [inline pkg/network/ebpf/c/runtime/conntrack.h:31]: Can we continue to use the status field for runtime compiled? Its less code.

Comment (akarpz) [reply pkg/network/ebpf/c/runtime/conntrack.h:31]: sure that makes sense, also we won't have the issues guessing for it

File: pkg/network/ebpf/c/runtime/conntrack.c [hunk 4]
```diff
@@ -61,7 +61,7 @@ int kprobe_ctnetlink_fill_info(struct pt_regs* ctx) {
         return 0;
     }
 
-    log_debug("kprobe/ctnetlink_fill_info: netns: %u, status: %x\n", get_netns(&ct->ct_net), status);
+    log_debug("kprobe/__nf_conntrack_hash_insert: netns: %u, status: %x\n", get_netns(&ct->ct_net), status);
```
Comment (hmahmood) [inline pkg/network/ebpf/c/runtime/conntrack.c:64]: this is `ctnetlink_fill_info`?

**Review by hmahmood — APPROVED** [review]
Just a small comment about incorrect function in `log_debug`

File: pkg/network/ebpf/c/runtime/conntrack.c (hunk 4 above)
Comment (akarpz) [reply pkg/network/ebpf/c/runtime/conntrack.c:64]: right yes

**Pull request merged**